extensions/sql-database-projects/src/dialogs/publishDatabaseQuickpick.ts

```diff
@@ -0,0 +1,152 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as constants from '../common/constants';
+import { IGenerateScriptSettings, IPublishSettings } from '../models/IPublishSettings';
+import { Project } from '../models/project';
+import { promptForPublishProfile } from './publishDatabaseDialog';
+
+/**
+ * Create flow for Publishing a database using only VS Code-native APIs such as QuickPick
+ */
+export async function launchPublishDatabaseQuickpick(project: Project): Promise<void> {
+
+	// 1. Select publish settings file (optional)
+	// TODO: Hook up to dacfx service
+	const browseProfileOption = await vscode.window.showQuickPick(
+		[constants.dontUseProfile, constants.browseForProfile],
+		{ title: constants.selectProfile, ignoreFocusOut: true });
+	if (!browseProfileOption) {
+		return;
+	}
+
+	// let publishSettingsFile: vscode.Uri | undefined;
+	if (browseProfileOption === constants.browseForProfile) {
+		const locations = await promptForPublishProfile(project.projectFolderPath);
+		if (!locations) {
+			return;
+		}
+		// publishSettingsFile = locations[0];
+	}
+
+	// 2. Select connection
+	// TODO: Hook up to MSSQL
+	const connectionProfile = await vscode.window.showQuickPick(
+		['Connection 1', 'Connection 2', 'Create New Connection'],
+		{ title: constants.selectConnection, ignoreFocusOut: true });
+	if (!connectionProfile) {
+		return;
+	}
+	const dbs = ['db1', 'db2'];
+	const dbQuickpicks = dbs.map(db => {
+		return {
+			label: db,
+			dbName: db
+		} as vscode.QuickPickItem & { dbName: string, isCreateNew?: boolean };
+	});
+	// Ensure the project name is an option, either adding it if it doesn't already exist or moving it to the top if it does
+	const projectNameIndex = dbs.findIndex(db => db === project.projectFileName);
+	if (projectNameIndex === -1) {
+		dbQuickpicks.unshift({ label: constants.newDatabaseTitle(project.projectFileName), dbName: project.projectFileName });
+	} else {
+		dbQuickpicks.splice(projectNameIndex, 1);
+		dbQuickpicks.unshift({ label: project.projectFileName, dbName: project.projectFileName });
+	}
+
+	dbQuickpicks.push({ label: constants.createNew, dbName: '', isCreateNew: true });
+	// 3. Select database
+	// TODO: Hook up to MSSQL
+	let databaseName = '';
+	while (databaseName === '') {
+		const selectedDatabase = await vscode.window.showQuickPick(
+			dbQuickpicks,
+			{ title: constants.selectDatabase, ignoreFocusOut: true });
+		if (!selectedDatabase) {
+			// User cancelled
+			return;
+		}
+		databaseName = selectedDatabase.dbName;
+		if (selectedDatabase.isCreateNew) {
+			databaseName = await vscode.window.showInputBox(
+				{
```

Comment:
nit: curly brace on previous line for TS

Reply:
Is this the standard for this extension? I don't think we have a standard (I don't pay attention to it at least as long as the linter is happy) and for me it makes more sense for object parameters like this.

Reply:
As best I can tell from the Google TS styleguide (the most official/comprehensive one out there, even though TS comes from us), it's standard for the language, so I've tried to adhere to it for the extension.  Plus, I notice you've done open-braces on the same line even for objects in this PR (lines 45, 97, 144), which is why the next-lined braces stood out.